scripts/helper/functions.sh

```diff
@@ -41,6 +41,9 @@ preflight_checks()
     verify_installed $cmd || exit 1
   done
 
+  # Verify Docker daemon is running
+  docker ps -q || exit 1
```

Comment:
@blaisdellk if it's just an exit, user may not know why. Please add an error message (see above/below) for examples.  Perhaps something like `ERROR: cp-demo requires Docker to be running.  Please start Docker and run again.`

Reply:
@blaisdellk makes sense.